website/docs/guides/catalog_subscription_and_sharing.html.markdown

```diff
@@ -0,0 +1,342 @@
+
+In **scenario 1**, the subscribing catalog will access immediately all items, and start synchronising them in background.
+In **scenario 2**, the subscribing catalog will get nothing, and will need to synchronise later on.
+In **scenario 3**, the subscribing catalog will access all items immediately, although it will have to be synchronised later on.
+
+### Subscription without automatic download 
+
+If we subscribe the catalog without automatic downloads (`make_local_copy = false`) there will be no immediate access to
+the catalog resources. The catalog and its items will need to be synchronised explicitly.
+When we are in this situation, adding `sync_catalog=true` to the creation script will only get general information about the
+catalog items, but not their data. It's like in the web interface: when we run "catalog sync", we see the list of vApp templates
+and media items, but the items are not available for usage.
+In the web interface, we would need to synchronise item by item. In `vcd_subscribed_catalog`, we can run several operations at once,
+i, e. we can choose among the following:
+
+1. `sync_catalog` will synchronise only the catalog, meaning that it will only download the **list of items**.
+2. `sync_all_media_items` and `sync_all_vapp_templates` will start the synchronisation of all the items that are visible
+    at that moment, i.e. the ones that were fetched with `sync_catalog`.
+3. `sync_vapp_templates` and `sync_media_items` will fetch only the items that we indicate.
+4. `sync_all` corresponds to `sync_catalogz + `sync_all_vapp_templates` + `sync_all_media_items`
+5. `sync_on_refresh` enables synchronisation every time that the catalog gets refreshed.
+
+Running a subscribed catalog without automatic downloads (`make_local_copy=false`) could be convenient when we know that
+the publishing catalog has a large number of items, and we don't need all of them, or we want to schedule the
+synchronisation during hours of light network traffic.
+
+The important thing to remember is that, when `synch_on_refresh` is not set, the synchronisation properties are only
+used during update, and due to the way Terraform works, they will be ignored if they were set also during creation.
+
+Thus, we have two ways of operating:
+
+* **in stages**, i.e. using none of the `sync_*` properties when creating the catalog, and then adding which ones we want
+  before running `terraform apply` a second time, but without setting `sync_on_refresh`. The synchronisation will only
+  happen from the second `apply`.
+* **all at once**, i.e. setting `sync_on_refresh` together with any of the options from 1. to 4. above. The synchronisation
+  will happen every time the resource is read, which may introduce some delays in the operations if you were expecting
+  `terraform refresh` to be quick. Despite the possible side effects, this is the recommended way.
```

Comment:
It says "all at once", but after reading it still may look like we don't get the synchronization via a single `apply`: it reads as if we need to do `apply` and then still at least one more `refresh` (or `plan`?). What about adding a note here and/or above in "5." bullet point that refresh happens automatically after an apply?

What I'm looking for is how to make the "implicit" sequence of operations clear to the users.

Reply:
The "all at once" means that the synchronisation is achieved through a single editing of the HCL script, as opposed to the "in stages", where the operations require multiple editing passes.
In both cases, you **may** need multiple Terraform operations to achieve the goal, either using `apply` or `refresh`.
The only case where you only need one operation is when the following circumstances concur:
* the subscribing catalog was created after the publishing catalog was filled
* the subscribing catalog has the option `make_local_copy = true`
and even in this case, if new items are added at the source we need either an apply or update to put the catalog up to date.